src/backend/distributed/sql/udfs/citus_get_node_clock/11.2-1.sql

```diff
@@ -1,6 +1,6 @@
 CREATE OR REPLACE FUNCTION pg_catalog.citus_get_node_clock()
     RETURNS pg_catalog.cluster_clock
-    LANGUAGE C STABLE PARALLEL SAFE STRICT
+    LANGUAGE C VOLATILE PARALLEL SAFE STRICT
```

Comment:
Semi-related: I think we should probably not mark as parallel safe given that we use sequences? Plus, seems unlikely that Postgres would need a parallel plan when we call this function

> Functions and aggregates must be marked PARALLEL UNSAFE if they write to the database, access sequences, change the transaction state even temporarily (e.g., a PL/pgSQL function that establishes an EXCEPTION block to catch errors), or make persistent changes to settings.

From pg docs: https://www.postgresql.org/docs/current/parallel-safety.html#:~:text=A%20parallel%20safe%20operation%20is,parallel%20query%20is%20in%20use.


ps: I know that we have LWLockAcquire(, ExclusiveLock), but still feels safer to define NOT PARALLEL SAFE?

Reply:
Both UDFs are intended to run independently and not embed in a query (in future we might have other usages). But having said that, I agree to this

>>Plus, seems unlikely that Postgres would need a parallel plan when we call this function

So, there is no need for us to declare this function as parallel safe. As we evolve and expand on this, we can revisit to see if this can be made parallel safe if needed, for now, I've changed it to parallel unsafe.